runtime/src/gpu/amd/gpu-amd.c

```diff
@@ -176,12 +176,22 @@ bool chpl_gpu_impl_is_device_ptr(const void* ptr) {
     }
   }
 
+#if ROCM_VERSION_MAJOR >= 6
+  // TODO: whats the correct expression here?
+  // https://rocm.docs.amd.com/projects/HIP/en/docs-6.0.0/doxygen/html/group___memory.html#ga7c3e8663feebb7be9fd3a1e5139bcefc
+  return res.type != hipMemoryTypeUnregistered;
```

Comment:
My theory is that `hipMemoryTypeUnregistered` is ROCm saying "you haven't allocated this memory through me or let me know about it". So, while this is true, I think it is only partially true as you can allocate host memory through HIP, which I believe will be "registered. I recommend keeping the previous checks here, too.

Reply:
On a quick look, I think we use this `is_device_ptr` check only for assertions and not for any logic.